### PR TITLE
s4803613: Classifying Alzheimer's Disease Using GFNet 

### PR DESCRIPTION
**Pull request by GavinSaiun**
This PR implements a classification model for Alzheimer's disease using the ADNI brain data, based on GFNet. The goal is to classify between Normal Control (NC) and Alzheimer's Disease (AD).

**Result:**
Final test set accuracy: 76.57%

**Changes Introduced**
The following files were added:
ReadMe.md: Description and summary of task
modules.py: Contains the GFNet architecture
dataset.py: Handles data loading, preprocessing, and data augmentation techniques for the ADNI dataset.
predict.py: Evaluates GFNet on testing set
train.py: Handles training of GFNet, saving model
Figures/: output images
slurm.txt: Driver script to run slurm file

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.7657. 

**Model and functions:**

- It correctly uses PyTorch to construct the GFNet models and functions.
- With data augmentation.

**Code design:** Good.

**Code comment and docstring:**

- Good code comments 
- Good function docstrings

**Difficulty:** Hard.

**Additional Comments:**

- Detailed experiment analysis in the report.
- intermediate numbers of commits.
- For a classification task, the confusion matrix would be better to show model performance on the test dataset than image plots.
- There should have been a conclusion in the report.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request could be more informative .


File Organizing: Well-organized files.

Commit Log:
Commit messages could be more progressive for the Recognition Problem.. 
Commits are regularly made.

Documentation:

Code comments and docstrings  are included.
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Can add about Conclusion and  future improvements in the ReadMe.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. Uses Data Aug</td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Some/Adequate Progressive commits. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-2</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Good work!
